types: BenchmarkCoinsAdditionIntersect panic

## Summary of Bug

`BenchmarkCoinsAdditionIntersect` panic.

## Version

1e33f39cba46f189ef44c9fe26f3f88a78a55780

## Steps to Reproduce

```
$ go test -run=NONE -bench=BenchmarkCoinsAdditionIntersect -benchmem
# github.com/keybase/go-keychain
cgo-gcc-prolog:203:11: warning: 'SecTrustedApplicationCreateFromPath' is deprecated: first deprecated in macOS 10.15 - No longer supported [-Wdeprecated-declarations]
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/System/Library/Frameworks/Security.framework/Headers/SecTrustedApplication.h:59:10: note: 'SecTrustedApplicationCreateFromPath' has been explicitly marked deprecated here
goos: darwin
goarch: arm64
pkg: github.com/cosmos/cosmos-sdk/types
BenchmarkCoinsAdditionIntersect/sizes:_A_1,_B_1-8         	28855777	        39.39 ns/op	      32 B/op	       1 allocs/op
BenchmarkCoinsAdditionIntersect/sizes:_A_5,_B_5-8         	 2903006	       449.5 ns/op	     520 B/op	      12 allocs/op
panic: Wrong argument: coins must be sorted

goroutine 12 [running]:
github.com/cosmos/cosmos-sdk/types.Coins.safeAdd(0x14000c7e080, 0x5, 0x5, 0x140001761e0, 0x14, 0x14, 0x7, 0x1402366aaf8, 0x2366aaf8)
	/Users/cuonglm/go/src/github.com/orijtech/cosmos-sdk/types/coin.go:290 +0x830
github.com/cosmos/cosmos-sdk/types.Coins.Add(...)
	/Users/cuonglm/go/src/github.com/orijtech/cosmos-sdk/types/coin.go:274
github.com/cosmos/cosmos-sdk/types.BenchmarkCoinsAdditionIntersect.func1.1(0x140005a2480)
	/Users/cuonglm/go/src/github.com/orijtech/cosmos-sdk/types/coin_benchmark_test.go:30 +0x2f4
testing.(*B).runN(0x140005a2480, 0x1)
	/Users/cuonglm/sdk/go1.16.7/src/testing/benchmark.go:192 +0xfc
testing.(*B).run1.func1(0x140005a2480)
	/Users/cuonglm/sdk/go1.16.7/src/testing/benchmark.go:232 +0x54
created by testing.(*B).run1
	/Users/cuonglm/sdk/go1.16.7/src/testing/benchmark.go:225 +0x68
exit status 2
FAIL	github.com/cosmos/cosmos-sdk/types	3.458s
```

____

## For Admin Use

- [ ] Not duplicate issue
- [ ] Appropriate labels applied
- [ ] Appropriate contributors tagged
- [ ] Contributor assigned/self-assigned

--- a/types/coin_benchmark_test.go
+++ b/types/coin_benchmark_test.go
@@ -6,7 +6,7 @@ import (
 )
 
 func coinName(suffix int) string {
-	return fmt.Sprintf("coinz%d", suffix)
+	return fmt.Sprintf("coinz%04d", suffix)
 }
 
 func BenchmarkCoinsAdditionIntersect(b *testing.B) {

